components/sup/src/manager.rs

```diff
@@ -1201,39 +1193,54 @@ impl Manager {
         None
     }
 
-    /// Return the Services that currently have a newer package in
-    /// Builder. These are removed from the internal `services` vec
-    /// for further transformation into futures.
+    /// Restart the Services that have an update or have set their `needs_restart` flag set.
     ///
     /// # Locking (see locking.md)
     /// * `RumorStore::list` (write)
     /// * `MemberList::entries` (read)
     /// * `RumorHeat::inner` (write)
     /// * `ManagerServices::inner` (write)
-    fn take_services_need_restart_rsw_mlr_rhw_msw(&mut self) -> Vec<Service> {
+    fn restart_services_rsw_mlr_rhw_msw(&mut self) {
         let service_updater = self.service_updater.lock();
 
         let mut state_services = self.state.services.lock_msw();
-        let mut idents_to_restart = Vec::new();
-        for (current_ident, service) in state_services.iter() {
-            if service.needs_restart {
-                idents_to_restart.push(current_ident.clone());
-            } else if let Some(new_ident) = service_updater.has_update(&service.service_group) {
-                outputln!("Updating from {} to {}", current_ident, new_ident);
+        let mut idents_to_restart_and_latest_desired_on_restart = Vec::new();
+        for (ident, service) in state_services.iter() {
+            if let Some(new_ident) = service_updater.has_update(&service.service_group) {
+                outputln!("Updating from {} to {}", ident, new_ident);
```

Comment:
It would be useful to indicate here whether we are actually rolling back to an older package, since we can compare the identifiers. Something as simple as changing the wording to "Rolling back to X from Y" would be all that's needed.

Reply:
Addressed in [79221](https://github.com/habitat-sh/habitat/pull/7513/commits/79221e07f7987043219a834e2b1f510f7848c0b9).